src/grid/grid3D.cpp

```diff
@@ -258,16 +258,22 @@ void Grid3D::Initialize(struct Parameters *P)
 #endif /*ROTATED_PROJECTION*/
 
 // Values for lower limit for density and temperature
+#ifdef TEMPERATURE_FLOOR
+  H.temperature_floor = P->temperature_floor;
+#else
+  H.temperature_floor = 0.0;
+#endif
+
 #ifdef DENSITY_FLOOR
-  H.density_floor = DENS_FLOOR;
+  H.density_floor = P->density_floor;
 #else
   H.density_floor = 0.0;
 #endif
 
-#ifdef TEMPERATURE_FLOOR
-  H.temperature_floor = TEMP_FLOOR;
+#ifdef SCALAR_FLOOR
+  H.scalar_floor = P->scalar_floor;
 #else
-  H.temperature_floor = 0.0;
+  H.scalar_floor = 0.0;
```

Comment:
While I understand this logic for the density and temperature floors, I think it doesn't actually apply for the scalar floor, since scalars can have negative values (think for example about the transverse velocities in the Riemann problem). In practice, it shouldn't matter, since you don't call the floor function unless you are applying a floor, in which case the value should have been specified in the parameter file. But I think it might make more sense rather than setting a default value of 0 here to just have the code throw a warning if the floor macro is turned on but no input value is supplied.